Enhance manual

alyst opened the pull request:
It is a subset of #18 that only updates the manual (the new API introduced by #18 is not discussed) and the docstrings:
- expand introduction clarifying Alexandria's prefixes
- move examples discussion to the relevant section
- convert `full: true` to `false` to condense the manual and better illustrate the numbering behavior
- tidy up sections numbering, disable Bibliographies numbering

SillyFreak commented on docs/manual.typ:
```diff
@@ -111,25 +134,29 @@ The example on the next page demonstrates some of these. If you find additional
-This approach is thus not suitable for multiple bibliographies that serve the same regions of a document. For this purpose, Alexandria also supports splitting the _loading_ and _rendering_ of a bibliography, giving you the opportunity to preprocess the bibliography entries. Instead of calling #ref-fn("bibliographyx()") directly, you'd use #ref-fn("load-bibliography()") followed by #ref-fn("get-bi ...
+In the previous example, the bibliographies were created for separate parts of a document, and each had its own independent numbering.
+This approach will not work when multiple bibliographies have to serve the same region of the document, because with overlapping numbers the citations become ambiguous.
+For this scenario, Alexandria allows decoupling _loading_ and _collection_ of the references from their _rendering_.
```
loading, collect*ing* and rendering

SillyFreak commented on docs/manual.typ:
```diff
@@ -142,45 +169,42 @@ An example could look like this:
-    // get the bibliography items
-    let (references, ..rest) =  get-bibliography("x-")
+    // get the bibliography items + additional information
+    let (references: bib_refs, ..bib_info) = get-bibliography("x-")
```
please use kebab-case for the identifiers here

SillyFreak commented on docs/manual.typ:
```diff
@@ -189,26 +213,33 @@ Note how all references are rendered once, although in a different presentation
+  #set heading(offset: 3, numbering: none)
   #context {
-    let (references, ..rest) =  get-bibliography("z-")
+    let (references: bib_refs, ..bib_info) = get-bibliography("z-")
```
here too

SillyFreak commented on src/lib.typ:
```diff
@@ -91,16 +85,16 @@
-  /// The prefix for which reference labels should be provided and citations should be processed.
+  /// the prefix for the citations within a group.
   /// -> string | auto
   prefix: auto,
```
Here and in the rest of the file, could you add a sentence like "this can be skipped if only one prefix is registered." to the `prefix` parameters? It wasn't there before, but it feels so missing now... thanks!

SillyFreak reviewed (commented):
looks great! I had some small remarks, but after they are addressed I'll merge this :)

(I'll review the other PR probably tomorrow)

alyst commented:
@SillyFreak Thank you for the review! I have addressed your comments. I've also further slightly tweaked some docstrings (see 8995490627a7c894816c72ade5f5106738289e40), I hope you are ok with these late changes.
